From ee0e5d1f6d458c8b86ca88df0579d422e353b55e Mon Sep 17 00:00:00 2001
From: Pardha Saradhi K <pardha.saradhi.kesapragada@intel.com>
Date: Mon, 14 Mar 2016 17:01:49 +0530
Subject: [PATCH 1905/2508] ASoC: Intel: Boards: Add probe dai links in IVI
 RSE Machine

commit b0d628e1bdb60135ab93ac8cfcf089964f9bf3ac from
https://github.com/01org/linux-apollolake-i

Add support for extractor and injector probe interfaces
for IVI RSE machine driver

Change-Id: Icffabaecc7c083aca280c1d078d47f0f6de2ad0f
Signed-off-by: Pardha Saradhi K <pardha.saradhi.kesapragada@intel.com>
Reviewed-on: https://git-gar-1.devtools.intel.com/gerrit/8398
Reviewed-by: Babu, Ramesh <ramesh.babu@intel.com>
Tested-by: Babu, Ramesh <ramesh.babu@intel.com>
---
 sound/soc/intel/boards/bxtp_ivi_rse_rt298.c |   28 +++++++++++++++++++++++++++
 1 files changed, 28 insertions(+), 0 deletions(-)

diff --git a/sound/soc/intel/boards/bxtp_ivi_rse_rt298.c b/sound/soc/intel/boards/bxtp_ivi_rse_rt298.c
index 445c87c..222d7f5 100644
--- a/sound/soc/intel/boards/bxtp_ivi_rse_rt298.c
+++ b/sound/soc/intel/boards/bxtp_ivi_rse_rt298.c
@@ -32,6 +32,8 @@ static const struct snd_soc_dapm_widget broxton_widgets[] = {
 	SND_SOC_DAPM_SPK("Speaker2", NULL),
 	SND_SOC_DAPM_SPK("Speaker3", NULL),
 	SND_SOC_DAPM_MIC("DMIC2", NULL),
+	SND_SOC_DAPM_SIGGEN("probe source"),
+	SND_SOC_DAPM_SINK("probe sink"),
 };
 
 static const struct snd_soc_dapm_route broxton_rt298_map[] = {
@@ -50,6 +52,8 @@ static const struct snd_soc_dapm_route broxton_rt298_map[] = {
 	{ "ssp1 Tx", NULL, "codec3_out"},
 	{ "hdmi_ssp0_in", NULL, "ssp0 Rx" },
 	{ "ssp0 Rx", NULL, "Dummy Capture1" },
+	{ "extractor cpr 8", NULL, "probe source"},
+	{ "probe sink", NULL, "injector cpr 9"},
 };
 
 static int bxtp_ssp0_gpio_init(struct snd_soc_pcm_runtime *rtd)
@@ -281,6 +285,30 @@ static struct snd_soc_dai_link broxton_rt298_dais[] = {
 		.capture_only = true,
 		.ignore_suspend = 1,
 	},
+        {
+                .name = "Bxt Probe playback",
+                .stream_name = "Probe Playback",
+                .cpu_dai_name = "Probe0 Pin",
+                .codec_name = "snd-soc-dummy",
+                .codec_dai_name = "snd-soc-dummy-dai",
+                .platform_name = "0000:00:0e.0",
+                .init = NULL,
+                .ignore_suspend = 1,
+                .nonatomic = 1,
+        },
+        {
+                .name = "Bxt Probe capture",
+                .stream_name = "Probe Capture",
+                .cpu_dai_name = "Probe1 Pin",
+                .codec_name = "snd-soc-dummy",
+                .codec_dai_name = "snd-soc-dummy-dai",
+                .platform_name = "0000:00:0e.0",
+                .init = NULL,
+                .ignore_suspend = 1,
+                .nonatomic = 1,
+        },
+
+
 
 	/* Back End DAI links */
 	{
-- 
1.7.5.4

